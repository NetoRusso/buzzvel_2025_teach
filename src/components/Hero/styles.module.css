
.hero {
  width: 100%;
  max-width: 1440px;
  height: auto;
  min-height: 100vh;
  padding: 16px;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
  gap: 40px;
  margin: 0 auto;
}

.hero_content_1 {
  width: calc(213.6px + 25.125vw);
  max-width: 575px;
  min-width: 294px;
  height: auto;
  min-height: 524px;
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-around;
  align-items: flex-start;
}

@media screen and (max-width: 1030px) {
  .hero_content_1 {
    margin-top: 96px;
  }
}

.hero_title {
  font-size: clamp(2.5rem, 1.7493rem + 3.0564vw, 4.5rem);
  line-height: 110%;
  font-weight: 800;
  position: relative;
}

.hero_title_trace { 
  width: calc(73.845271255px + 7.92741165vw);
  max-width: 190.5px;
  min-width: 108px;
  height: auto;
  position: absolute;
  top: clamp(1.875rem, 1.1712rem + 2.8653vw, 3.75rem);
  left: 5px;
  z-index: -1;
}

.hero_paragraph { 
  width: 80%;
  min-width: 294px;
  font-size: clamp(1rem, 1rem + 0vw, 1rem);
  line-height: 160%;
  font-weight: 400;
}

.hero_btns { 
  width: 100%;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: flex-start;
  gap: 24px;
}

.hero_btn {
  width: 41%;
  max-width: 239px;
  min-width: 153px;
  height: calc(41.9942706318px + 1.52817574vw);
  border-radius: 8px;
  background-color: var(--orange-600);
  border: none;
  font-size: clamp(1rem, 0.8123rem + 0.7641vw, 1.5rem);
  font-weight: 700;
  color: var(--background-white);
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.hero_btn:hover {
  scale: 0.95;
  background-color: var(--background-white);
  color: var(--blue-Gray-900);
  border: 2px solid var(--blue-Gray-900);
}

.hero_demo {
  width: calc(149.1174784062px + 2.00573066vw);
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: center;
  gap: 8px;
  font-size: clamp(1rem, 0.9062rem + 0.382vw, 1.25rem);
  color: var(--blue-600);
  transition: all 0.3s ease-in-out;
}

.hero_demo:hover {
  scale: 0.95;
  opacity: 0.9;
}

.hero_companies {
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: flex-start;
  gap: 16px;
  margin-top: 48px;
}

.hero_companies_text {
  width: auto;
  min-width: 135px;
  text-align: left;
  color: var(--blue-Gray-600);
}

.hero_companies_logos { 
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: space-around;
  gap: 24px;
}

.hero_content_2 {
  width: calc(206.571428576px + 27.32142857vw);
  max-width: 625px;
  min-width: 294px;

}

.hero_image {
  width: 100%;
  height: auto;
}