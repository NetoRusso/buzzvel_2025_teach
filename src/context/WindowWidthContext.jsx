'use client'
import { createContext, useState, useEffect } from "react";

const WindowWidthContext = createContext();

export const WindowWidthProvider = ({ children }) => {
  const [windowWidth, setWindowWidth] = useState(0);
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    const handleResize = () => {
      setWindowWidth(window.innerWidth);
    }

    window.addEventListener("resize", handleResize);

    handleResize();

    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  useEffect(() => {
    if(windowWidth < 840) {
      setIsMobile(true);

    } else if(windowWidth > 840) {
      setIsMobile(false);

    }
  }, [windowWidth]);

  return (
    <WindowWidthContext.Provider value={{ windowWidth, isMobile }}>
      {children}
    </WindowWidthContext.Provider>
  );
};

export { WindowWidthContext };