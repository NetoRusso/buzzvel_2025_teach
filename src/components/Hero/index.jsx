
import classNames from "classnames";
import styles from "./styles.module.css";
import Link from "next/link";
import Image from "next/image";
import { WindowWidthContext } from "@/context/WindowWidthContext";
import { useContext } from "react";

import trace from "@/assets/SVG/trace.svg";
import play from "@/assets/SVG/play_symbol.svg";
import image_hero from "@/assets/Group_hero.png";

import companies01 from "@/assets/SVG/companies01.svg";
import companies02 from "@/assets/SVG/companies02.svg";
import companies03 from "@/assets/SVG/companies03.svg";
import companies04 from "@/assets/SVG/companies04.svg";
import companies05 from "@/assets/SVG/companies05.svg";



const Hero = () => {

  const { isMobile } = useContext(WindowWidthContext);

  return (
    <section className={styles.hero} aria-label="Overview of Teach: Education for students worldwide, sign-up and demo options, plus our leading partners.">
      <div className={classNames({
        [styles.hero_content]: true,
        [styles.hero_content_1]: true,
      })}>
        <h1 className={styles.hero_title}>
          Teach students worldwide
          <Image className={styles.hero_title_trace} src={trace} alt="Trace" width={191} height={13} loading="lazy" />
        </h1>
        <p className={styles.hero_paragraph}>
          Amet nunc diam orci duis ut sit diam arcu, nec. Eleifend proin massa tincidunt viverra lectus pulvinar. Nunc ipsum est pellentesque turpis ultricies.
        </p>
        <div className={styles.hero_btns}>
          <button className={styles.hero_btn}>
            Sign Up Now
          </button>
          <Link href="#" className={styles.hero_demo}>
            <Image className={styles.hero_demo_play} src={play} alt="Play" width={24} height={24} loading="lazy"/>
            View Demo
          </Link>
        </div>
        <div className={styles.hero_companies}>
          {isMobile ? (<p className={styles.hero_companies_text}>Trusted by leading companies</p>):(<p className={styles.hero_companies_text}>Trusted by<br />leading companies</p>)}
          <div className={styles.hero_companies_logos}>
            <Image className={styles.companies_logo_img} src={companies01} alt="companies01" width={32} height={32} loading="lazy"/>
            <Image className={styles.companies_logo_img} src={companies02} alt="companies02" width={32} height={32} loading="lazy"/>
            <Image className={styles.companies_logo_img} src={companies03} alt="companies03" width={32} height={32} loading="lazy"/>
            <Image className={styles.companies_logo_img} src={companies04} alt="companies04" width={32} height={32} loading="lazy"/>
            <Image className={styles.companies_logo_img} src={companies05} alt="companies05" width={32} height={32} loading="lazy"/>
          </div>
        </div>
      </div>

      <div className={classNames({
        [styles.hero_content]: true,
        [styles.hero_content_2]: true,
      })}>
        <Image className={styles.hero_image} src={image_hero} alt="image_hero" width={625} height={545} loading="lazy" />
      </div>
    </section>
  )
};

export default Hero;