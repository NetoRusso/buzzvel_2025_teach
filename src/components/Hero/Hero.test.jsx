import { render, screen } from '@testing-library/react';
import Hero from './index';
import { WindowWidthContext } from '@/context/WindowWidthContext';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props) => {
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
    const { priority, ...rest } = props;
    return <img {...rest} data-priority={priority ? 'true' : 'false'} />;
  },
}));

jest.mock('next/link', () => {
  // eslint-disable-next-line react/display-name
  return ({children, href, ...rest}) => {
    return <a href={href} {...rest}>{children}</a>;
  }
});

const renderWithContext = (ui, { providerProps, ...renderOptions }) => {
  return render(
    <WindowWidthContext.Provider value={providerProps}>{ui}</WindowWidthContext.Provider>,
    renderOptions
  );
};

describe('Hero Component', () => {
  const desktopContext = { windowWidth: 1024, isMobile: false, setWindowWidth: jest.fn() };
  const mobileContext = { windowWidth: 400, isMobile: true, setWindowWidth: jest.fn() };

  it('renders the main title and paragraph', () => {
    renderWithContext(<Hero />, { providerProps: desktopContext });
    expect(screen.getByRole('heading', { name: /Teach students worldwide/i, level: 1 })).toBeInTheDocument();
    expect(screen.getByText(/Amet nunc diam orci duis ut sit diam arcu, nec./i)).toBeInTheDocument();
  });

  it('renders the action buttons', () => {
    renderWithContext(<Hero />, { providerProps: desktopContext });
    expect(screen.getByRole('button', { name: /Sign Up Now/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /View Demo/i })).toBeInTheDocument();
    expect(screen.getByAltText('Play')).toBeInTheDocument();
  });

  it('renders the main hero image with priority', () => {
    renderWithContext(<Hero />, { providerProps: desktopContext });
    const heroImage = screen.getByAltText('image_hero');
    expect(heroImage).toBeInTheDocument();
    expect(heroImage).toHaveAttribute('data-priority', 'true');
  });

  it('renders company logos', () => {
    renderWithContext(<Hero />, { providerProps: desktopContext });
    expect(screen.getByAltText('companies01')).toBeInTheDocument();
    expect(screen.getByAltText('companies02')).toBeInTheDocument();
    expect(screen.getByAltText('companies03')).toBeInTheDocument();
    expect(screen.getByAltText('companies04')).toBeInTheDocument();
    expect(screen.getByAltText('companies05')).toBeInTheDocument();
  });

  describe('Trusted by companies text', () => {
    it('renders desktop version of "Trusted by" text when not mobile', () => {
      renderWithContext(<Hero />, { providerProps: desktopContext });
      const trustedTextElement = screen.getByText((content, node) => {
        const hasText = (node) => node.textContent === "Trusted byleading companies";
        const nodeHasText = hasText(node);
        const childrenDontHaveText = Array.from(node.children).every(
          (child) => !hasText(child)
        );
        return nodeHasText && childrenDontHaveText;
      });
      expect(trustedTextElement).toBeInTheDocument();
      expect(trustedTextElement.querySelector('br')).toBeInTheDocument();
    });

    it('renders mobile version of "Trusted by" text when mobile', () => {
      renderWithContext(<Hero />, { providerProps: mobileContext });
      expect(screen.getByText('Trusted by leading companies')).toBeInTheDocument();
      const trustedTextElement = screen.getByText('Trusted by leading companies');
      expect(trustedTextElement.querySelector('br')).not.toBeInTheDocument();
    });
  });
  
  it('renders the decorative trace image for the title', () => {
    renderWithContext(<Hero />, { providerProps: desktopContext });
    expect(screen.getByAltText('Trace')).toBeInTheDocument();
  });

  it('matches snapshot on desktop', () => {
    const { container } = renderWithContext(<Hero />, { providerProps: desktopContext });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot on mobile', () => {
    const { container } = renderWithContext(<Hero />, { providerProps: mobileContext });
    expect(container.firstChild).toMatchSnapshot();
  });
});