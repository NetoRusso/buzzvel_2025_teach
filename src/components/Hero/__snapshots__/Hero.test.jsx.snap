// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches snapshot on desktop 1`] = `
<section
  aria-label="Overview of Teach: Education for students worldwide, sign-up and demo options, plus our leading partners."
  class="hero"
>
  <div
    class="hero_content hero_content_1"
  >
    <h1
      class="hero_title"
    >
      Teach students worldwide
      <img
        alt="Trace"
        class="hero_title_trace"
        data-priority="false"
        height="13"
        loading="lazy"
        src="[object Object]"
        width="191"
      />
    </h1>
    <p
      class="hero_paragraph"
    >
      Amet nunc diam orci duis ut sit diam arcu, nec. Eleifend proin massa tincidunt viverra lectus pulvinar. Nunc ipsum est pellentesque turpis ultricies.
    </p>
    <div
      class="hero_btns"
    >
      <button
        class="hero_btn"
      >
        Sign Up Now
      </button>
      <a
        class="hero_demo"
        href="#"
      >
        <img
          alt="Play"
          class="hero_demo_play"
          data-priority="false"
          height="24"
          loading="lazy"
          src="[object Object]"
          width="24"
        />
        View Demo
      </a>
    </div>
    <div
      class="hero_companies"
    >
      <p
        class="hero_companies_text"
      >
        Trusted by
        <br />
        leading companies
      </p>
      <div
        class="hero_companies_logos"
      >
        <img
          alt="companies01"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies02"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies03"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies04"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies05"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
      </div>
    </div>
  </div>
  <div
    class="hero_content hero_content_2"
  >
    <img
      alt="image_hero"
      class="hero_image"
      data-priority="true"
      height="545"
      src="[object Object]"
      width="625"
    />
  </div>
</section>
`;

exports[`Hero Component matches snapshot on mobile 1`] = `
<section
  aria-label="Overview of Teach: Education for students worldwide, sign-up and demo options, plus our leading partners."
  class="hero"
>
  <div
    class="hero_content hero_content_1"
  >
    <h1
      class="hero_title"
    >
      Teach students worldwide
      <img
        alt="Trace"
        class="hero_title_trace"
        data-priority="false"
        height="13"
        loading="lazy"
        src="[object Object]"
        width="191"
      />
    </h1>
    <p
      class="hero_paragraph"
    >
      Amet nunc diam orci duis ut sit diam arcu, nec. Eleifend proin massa tincidunt viverra lectus pulvinar. Nunc ipsum est pellentesque turpis ultricies.
    </p>
    <div
      class="hero_btns"
    >
      <button
        class="hero_btn"
      >
        Sign Up Now
      </button>
      <a
        class="hero_demo"
        href="#"
      >
        <img
          alt="Play"
          class="hero_demo_play"
          data-priority="false"
          height="24"
          loading="lazy"
          src="[object Object]"
          width="24"
        />
        View Demo
      </a>
    </div>
    <div
      class="hero_companies"
    >
      <p
        class="hero_companies_text"
      >
        Trusted by leading companies
      </p>
      <div
        class="hero_companies_logos"
      >
        <img
          alt="companies01"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies02"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies03"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies04"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
        <img
          alt="companies05"
          class="companies_logo_img"
          data-priority="false"
          height="32"
          loading="lazy"
          src="[object Object]"
          width="32"
        />
      </div>
    </div>
  </div>
  <div
    class="hero_content hero_content_2"
  >
    <img
      alt="image_hero"
      class="hero_image"
      data-priority="true"
      height="545"
      src="[object Object]"
      width="625"
    />
  </div>
</section>
`;
